kratos/geometries/quadrilateral_2d_4.h

```diff
@@ -1040,6 +1066,210 @@ template<class TPointType> class Quadrilateral2D4
+        if (!AxisTest_X(e0[1],e0[2],fey,fez,v0,v2,v3,boxhalfsize)) return false;
+        if (!AxisTest_Y(e0[0],e0[2],fex,fez,v0,v2,v3,boxhalfsize)) return false;
+      //if (!AxisTest_Z(e0[1],e0[0],fey,fez,v0,v2,v3,boxhalfsize)) return false;
+        
+        fex = fabs(e1[0]);
+        fey = fabs(e1[1]);
+        fez = fabs(e1[2]);
+        if (!AxisTest_X(e1[1],e1[2],fey,fez,v1,v3,v0,boxhalfsize)) return false;
+        if (!AxisTest_Y(e1[0],e1[2],fex,fez,v1,v3,v0,boxhalfsize)) return false;
+      //if (!AxisTest_Z(e0[1],e0[0],fey,fez,v1,v3,v0,boxhalfsize)) return false;
+        
+        fex = fabs(e2[0]);
+        fey = fabs(e2[1]);
+        fez = fabs(e2[2]);
+        if (!AxisTest_X(e2[1],e2[2],fey,fez,v2,v0,v1,boxhalfsize)) return false;
+        if (!AxisTest_Y(e2[0],e2[2],fex,fez,v2,v0,v1,boxhalfsize)) return false;
+      //if (!AxisTest_Z(e0[1],e0[0],fey,fez,v2,v0,v1,boxhalfsize)) return false;
+        
+        fex = fabs(e3[0]);
+        fey = fabs(e3[1]);
+        fez = fabs(e3[2]);
+        if (!AxisTest_X(e3[1],e3[2],fey,fez,v3,v1,v2,boxhalfsize)) return false;
+        if (!AxisTest_Y(e3[0],e3[2],fex,fez,v3,v1,v2,boxhalfsize)) return false;
+      //if (!AxisTest_Z(e0[1],e0[0],fey,fez,v3,v1,v2,boxhalfsize)) return false;
+        
+        
+        /* Bullet 1:  */
+        /*  first test overlap in the {x,y,z}-directions */
+        /*  find min, max of the quadrilateral for each direction, and test for  */
+        /*  overlap in that direction -- this is equivalent to testing a minimal */
+        /*  AABB around the triangle against the AABB */
+
+        /* test in X-direction */
+        FindMinMax(v0[0],v1[0],v2[0],v3[0],min,max);
+        if(min>boxhalfsize[0] || max<-boxhalfsize[0]) return false;
+        
+        /* test in Y-direction */
+        FindMinMax(v0[1],v1[1],v2[1],v3[1],min,max);
+        if(min>boxhalfsize[1] || max<-boxhalfsize[1]) return false;
+        
+        /* test in Z-direction */
+        FindMinMax(v0[2],v1[2],v2[2],v3[2],min,max);
+        if(min>boxhalfsize[2] || max<-boxhalfsize[2]) return false;
+        
+
+        /* Bullet 2: */
+        /*  test if the box intersects the plane of the triangle */
+        /*  compute plane equation of quadrilateral: normal*x+d=0 */
+        MathUtils<double>::CrossProduct(normal, e0, e1);
+        d =- inner_prod(normal,v0);  /* plane eq: normal*x+d=0 */
+        if(!PlaneBoxOverlap(normal,d,boxhalfsize)) return false;
+        
+        return true;  /* box and triangle overlaps */
+    }
+    
+    
+
+//*************************************************************************************
+//*************************************************************************************
+
+
+    void FindMinMax(const double& x0,
+                    const double& x1,
+                    const double& x2,
+                    double& min,
+                    double& max)
+    {
+        min = max = x0;
+        if(x1<min) min=x1;
```

Comment:
I am not sure, but maybe is far more efficient to use cases or else if insted of if

Reply:
Or even:

~~~cpp
min = x1 < min ? x1 : x2 < min ? x2 : min;
max = x1 > max ? x1 : x2 > max ? x2 : max;
~~~

Reply:
I have replaced some _if_ with _else_ and _else if_ to make it more efficient